test/config-luatest/config_test.lua

```diff
@@ -2034,3 +2034,45 @@ g.test_do_no_revoke_user_privs = function(g)
         t.assert_equals(actual_privs, exp)
     end)
 end
+
+-- This scenario has been broken before gh-10855.
+-- Before gh-10855 config:get() used to change the passed path
+-- if it's been passed as a table.
+g.test_get_not_changes_path = function(g)
+    local dir = treegen.prepare_directory({}, {})
+    local config = [[
+        credentials:
+          users:
+            guest:
+              roles:
+              - super
+
+        iproto:
+          listen:
+            - uri: unix/:./{{ instance_name }}.iproto
+
+        groups:
+          group-001:
+            replicasets:
+              replicaset-001:
+                instances:
+                  instance-001: {}
+    ]]
+    local config_file = treegen.write_file(dir, 'config.yaml', config)
+    local opts = {
+        config_file = config_file,
+        chdir = dir,
+        alias = 'instance-001',
+    }
+    g.server = server:new(opts)
+    g.server:start()
```

Comment:
Can be replaced with `helpers.success_case`. Less code -- easier to read.

Reply:
Applied. Thank you.